### DISCUSSION TITLE
[Bug] "Internal Error" When Querying Product Variant StoreAvailability On The Storefront API

### DISCUSSION THREAD

**Discussion by huykon** [Shopify/storefront-api-feedback#205] [Help]
**Storefront API Version**: 2023-07.

**What I'm trying to do**: Query `storeAvailability` data associated with a product variant.

the query:
```
query getStoreAvailability($handle: String!) {
    product(handle: $handle) {
      availableForSale
      variantBySelectedOptions(selectedOptions: {name: "Color", value: "Ice"}) {
        sku
        title
        weight
        quantityAvailable
        storeAvailability(first: 1) {
          edges {
            node {
              available
              pickUpTime
              location {
                name
              }
            }
          }
        }

      }
    }
  }
``` 

which returned:
```
Unexpected Server Error

Error: Internal error. Looks like something went wrong on our end.
Request ID: e49c46aa-80d3-4815-9e45-68e8acf8da1b (include this in support requests).
``` 

Is this a bug in the Storefront API, or is there another issue at play?
I also create a ticket here: https://community.shopify.com/c/hydrogen-headless-and-storefront/quot-internal-error-quot-when-querying-product-variant/m-p/2194789#M8209

**Comment by johncraigcole** [comment]
Hey @huykon 👋 . Not seeing any issues with the query, against my own store. 

One other thing I might try just to see if it makes a difference, is using the `near` attribute on `storeAvailability`. `near` takes a `latitude` and `longitude` and I would be curious if you use a coordinate close to one of the pick up locations if it made a difference.

If that doesn't help, and you've done all that Liam suggested on the forum post, it might be worth contacting support so that we can have you authenticated on the store in question, and be able to access the store data. If you share the ticket number here, I can try and take a closer look.
